src/components/bottom-control-buttons/__tests__/usePlayBackRateButtonHook.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import { useMediaStore } from '../../../context';
+import { setupMediaProvider, userEvent } from '../../../utils/testing-render';
+import {
+	usePlaybackRateButtonHook,
+	UsePlaybackRateButtonHook,
+} from '../hooks/usePlaybackRateButtonHook';
+
+const playbackRates = [1, 2];
+const TEST_ID = 'test-id';
+
+function setupPlaybackRate<T extends number>(playbackRates: T[], initial?: T) {
+	const clickHandler = {} as UsePlaybackRateButtonHook;
+	function NullComponent() {
+		const [playbackRate, setPlaybackRate] = useMediaStore(state => [
+			state.playbackRate,
+			state.setPlaybackRate,
+		]);
+		const { handleClick } = usePlaybackRateButtonHook({
+			currentRate: initial ?? playbackRate,
+			setPlaybackRate,
+			playbackRates,
+		});
+		Object.assign(clickHandler, handleClick);
+
+		return <button data-testid={TEST_ID} onClick={handleClick} />;
+	}
+
+	const setupResults = setupMediaProvider(<NullComponent />);
+	return { ...setupResults, ...clickHandler };
+}
+
+describe('usePlaybackRateButtonHook', () => {
+	describe('initialization', () => {
+		it('currentRate do not belongs to playbackRates', async () => {
+			const { handleClick } = setupPlaybackRate(playbackRates, 0);
+
+			expect(() => {
+				handleClick();
+			}).toThrowError();
+		});
+		it('default', () => {
```

Comment:
Improve wording (so one can understand the test when looking through the test report): `defaults currentRate to first playback value`